### PR TITLE
Allow the DI container to set a callable.

### PR DESCRIPTION
**Pull request by eddieajau**

### CODE REVIEW

File: src/Joomla/DI/Container.php [hunk 1]
```diff
@@ -277,7 +277,7 @@ public function set($key, $value, $shared = false, $protected = false)
 		}
 
 		// If the provided $value is not a closure, make it one now for easy resolution.
-		if (!($value instanceof \Closure))
+		if (!($value instanceof \Closure) && !is_callable($value))
```
Comment (dongilbert) [inline src/Joomla/DI/Container.php]: I would change this line to just

``` php
if (! is_callable($value))
```

File: src/Joomla/DI/Container.php [hunk 2]
```diff
@@ -355,7 +355,7 @@ public function get($key, $forceNew = false)
 			return $this->instances[$key];
 		}
 
-		return $raw['callback']($this);
+		return ($raw['callback'] instanceof \Closure) ? $raw['callback']($this) : call_user_func($raw['callback'], $this);
```
Comment (dongilbert) [inline src/Joomla/DI/Container.php]: And this line to just 

``` php
return call_user_func($raw['callback'], $this);
```

**Comment by piotr-cz** [comment]
cool :+1:

**Comment by eddieajau** [comment]
Fixed. Thanks!

**Comment by dongilbert** [comment]
Thank you - it definitely makes things more testable.

**Pull request merged**